File: Computer Science /FCS-BWL-1/calculate_pi.cpp
// Calculation of Pi in C++ according to the Leibniz rule

#include <iostream>
#include <cmath>
#include <chrono>

using namespace std;

double calcPi(int n) {
    double pi = 0;
	double numer = 4.0;
    for(int i = 0; i < n ; i++) {
		double denom = (2*i+1);
		double term = numer/denom;
		if (i%2 == 1) pi = pi - term;
		else pi = pi + term;
	}
	return pi;
}

int main() {
	int terms = 1000;

    cout.setf(ios::fixed);
    cout.setf(ios::showpoint);
    cout.precision(20);

	chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
	double pi = calcPi(terms);
	int duration = chrono::duration_cast<chrono::milliseconds>(std::chrono::steady_clock::now() - begin).count();
	
	cout << "PI: " << pi << endl;
	cout << "Elapsed Time (C++): " << duration << "ms" << endl;
}

File: Computer Science /FCS-BWL-1/calculate_pi.py
# Calculation of Pi in python according to the Leibniz rule

import time

def calcPi(n):
    pi,numer = 0,4.0
    for i in range(n):
        denom = (2*i+1)
        term = numer/denom
        if (i%2 == 1):
            pi -= term
        else:
            pi += term
    return(pi)

terms = 1000

start_time = time.time()
pi = calcPi(terms)
elapsed_time = (time.time() - start_time) * 1000

print("PI: ", pi)
print("Elapsed Time (Python): ", elapsed_time, "ms")